light/lightchain.go

```diff
@@ -501,10 +502,26 @@ func (lc *LightChain) GetHeaderByNumberOdr(ctx context.Context, number uint64) (
 	return GetHeaderByNumber(ctx, lc.odr, number)
 }
 
-func (self *LightChain) GetVMConfig() *vm.Config {
+func (lc *LightChain) GetVMConfig() *vm.Config {
 	return &vm.Config{}
 }
 
+// NewEVMRunner creates the System's EVMRunner for given header & sttate
+func (lc *LightChain) NewEVMRunner(header *types.Header, state vm.StateDB) vm.EVMRunner {
+	return vmcontext.NewEVMRunner(lc, header, state)
+}
+
+// NewEVMRunnerForCurrentBlock creates the System's EVMRunner for current block & state
+func (lc *LightChain) NewEVMRunnerForCurrentBlock() (vm.EVMRunner, error) {
+	header := lc.CurrentHeader()
+	// FIXME small race condition here. Need to make sure state matches header
```

Comment:
I believe we can avoid the race condition by doing:
```
state, err := NewState(context.Background(), header, bc.odr), nil
```

Essentially, doing the same as what `lc.State()` does except with this header rather than have it call `lc.CurrentHeader()` again.